src/test/java/org/jenkins/plugins/lockableresources/DeclarativePipelineTest.java

```diff
@@ -0,0 +1,139 @@
+    j.waitForCompletion(b1);
+    j.assertBuildStatus(Result.SUCCESS, b1);
+    j.assertLogContains("Resource [resource1] did not exist. Created.", b1);
+    assertNull(LockableResourcesManager.get().fromName("resource1"));
+  }
+
+  @Test
+  public void lockByLabelInOptionsSection() throws Exception {
+    LockableResourcesManager.get().createResourceWithLabel("resource1", "label1");
+    WorkflowJob p = j.createProject(WorkflowJob.class, "p");
+    p.setDefinition(
+      new CpsFlowDefinition(m("pipeline {",
+        " agent none",
+        " options {",
+        "  lock label: 'label1'",
+        " }",
+        " stages {",
+        "  stage('test') {",
+        "   steps {",
+        "    echo 'foo'",
+        "   }",
+        "  }",
+        " }",
+        "}"), true));
+    WorkflowRun b1 = p.scheduleBuild2(0).waitForStart();
+    j.waitForCompletion(b1);
+    j.assertBuildStatus(Result.SUCCESS, b1);
+    j.assertLogContains("Lock acquired on [Label: label1]", b1);
+  }
+
+  @Test
+  public void stepScriptLockByLabel() throws Exception {
+    LockableResourcesManager.get().createResourceWithLabel("resource1", "label1");
+    WorkflowJob p = j.createProject(WorkflowJob.class, "p");
+    p.setDefinition(
+      new CpsFlowDefinition(m("pipeline {",
+        " agent none",
+        " stages {",
+        "  stage('test') {",
+        "   steps {",
+        "    script {",
+        "       lock(label: 'label1', variable: 'LABEL_LOCKED', quantity: 1) {",
+        "         echo \"Lock acquired: ${LABEL_LOCKED}\"",
+        "       }",
+        "     }",
+        "   }",
+        "  }",
+        " }",
+        "}"), true));
+    WorkflowRun b1 = p.scheduleBuild2(0).waitForStart();
+    j.waitForCompletion(b1);
+    j.assertBuildStatus(Result.SUCCESS, b1);
+    j.assertLogContains("Lock acquired: resource1", b1);
+  }
+
+  @Test
+  public void stepLockByLabel() throws Exception {
+    LockableResourcesManager.get().createResourceWithLabel("resource1", "label1");
+    WorkflowJob p = j.createProject(WorkflowJob.class, "p");
+    p.setDefinition(
+      new CpsFlowDefinition(m("pipeline {",
+        " agent none",
+        " stages {",
+        "  stage('test') {",
+        "   steps {",
+        "     lock(label: 'label1', variable: 'LABEL_LOCKED', quantity: 1) {",
+        "       echo \"Lock acquired: ${LABEL_LOCKED}\"",
+        "     }",
+        "   }",
+        "  }",
+        " }",
+        "}"), true));
+    WorkflowRun b1 = p.scheduleBuild2(0).waitForStart();
+    j.waitForCompletion(b1);
+    j.assertBuildStatus(Result.SUCCESS, b1);
+    j.assertLogContains("Lock acquired: resource1", b1);
+  }
+
+  @Test
+  public void missingLabel() throws Exception {
+    WorkflowJob p = j.createProject(WorkflowJob.class, "p");
+    p.setDefinition(
+      new CpsFlowDefinition(m("pipeline {",
+        " agent none",
+        " stages {",
+        "  stage('test') {",
+        "   steps {",
+        "     lock() {",
```

Comment:
This is useless. I think in this case it shall produce a exception.

Reply:
Agreed. I think this has to do with changes in the meantime. I will check.

Reply:
So, currently, the plugin does not validate/enforce that one of label or resource is specified.
I can either remove the test or add a validation that raises an error if nothing is specified

Reply:
It does. Bur only in script generator and in freestyle project.

Please keep the test as is and create a follow issue. Then we can create new PR with the check. 
Because the change with exception might break some pipelines by end users as well.  In that case we can write proper change log and in worst case we can revert changes only from one PR.